vendor/github.com/docker/libnetwork/drivers/overlay/joinleave.go

```diff
@@ -145,6 +145,21 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 	return nil
 }
 
+func (d *driver) DecodeTableEntry(tablename string, key string, value []byte) (string, string) {
+	if tablename != ovPeerTable {
+		logrus.Errorf("DecodeTableEntry: unexpected table name %s", tablename)
+		return "", ""
+	}
+
+	var peer PeerRecord
+	if err := proto.Unmarshal(value, &peer); err != nil {
+		logrus.Errorf("DecodeTableEntry: failed to unmarshal peer record for key %s: %v", key, err)
+		return "", ""
+	}
+
+	return key, "Host IP: " + peer.TunnelEndpointIP
```

Comment:
Sorry for confusion, what I meant was converting a human-friendly string returned as the 2nd retvalue here to a machine-friendly map.
i.e.
```go
return key, map[string]string{"HostIP": peer.TunnelEndpointIP}
```